main/src/ca/uwaterloo/flix/language/phase/unification/ClassEnvironment.scala

```diff
@@ -111,11 +112,17 @@ object ClassEnvironment {
   }
 
   /**
-    * Returns the list of constraints that hold if the given constraint `tconstr` hold, using the superclasses of the constraint.
-    * (Currently only returns the given constraint since superclasses are not yet implemented.)
+    * Returns the list of constraints that hold if the given constraint `tconstr` holds, using the superclasses of the constraint.
     */
-  private def bySuper(tconstr: Ast.TypeConstraint)(implicit flix: Flix): List[Ast.TypeConstraint] = {
-    List(tconstr)
+  private def bySuper(tconstr: Ast.TypeConstraint, classEnv: Map[Symbol.ClassSym, Ast.ClassContext])(implicit flix: Flix): List[Ast.TypeConstraint] = {
```

Comment:
Can you add a  bit more documentation? Maybe put an example in the doc for the method and few more comments inside.